Documentation/gettingstarted/egress-gateway.rst

```diff
@@ -0,0 +1,174 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _egress_gateway:
+
+**********************
+Egress Gateway (beta)
+**********************
+
+Egress gateway provides a way to configure the egress policy. Without
+this feature, pod traffic could only egress from the node it is running
+on and masqueraded with the node IP.
+
+This document explains how to configure Cilium to enable egress gateway
+as well as how to configure egress NAT policy to route and SNAT the egress
+traffic for a specific workload.
+
+.. include:: ../beta.rst
+
+Configure Cilium
+================
+
+.. include:: k8s-install-download-release.rst
+
+The Cilium Egress Gateway feature relies on :ref:`kubeproxy-free`,
+follow the guide to create a new deployment. The beta feature is disabled by default.
+Enable the feature by setting the ``localRedirectPolicy`` value to ``true``.
+
+.. parsed-literal::
+
+   helm install cilium |CHART_RELEASE| \\
+     --set enableEgressGateway=true
+
+Verify that Cilium agent pod is running.
+
+.. code-block:: bash
+
+    $ kubectl -n kube-system get pods -l k8s-app=cilium
+    NAME           READY   STATUS    RESTARTS   AGE
+    cilium-5ngzd   1/1     Running   0          3m19s
+
+Validate that the Cilium Egress NAT Policy CRD has been registered.
+
+.. code-block:: bash
+
+	   $ kubectl get crds
+	   NAME                              CREATED AT
+	   [...]
+	   ciliumegressnatpolicy.cilium.io   2021-03-26T02:11:21Z
+
+
+Optionally, Egress Gateway support can be enabled by setting
+``enable-egress-gateway`` to ``true`` in ConfigMap. By default, it is disabled.
+
+To set ``enable-egress-gateway`` to ``true``, it requires
+``enable-bpf-masqurade`` and ``kube-proxy-replacement``. It cannot be enabled
+with these 2 options set as ``false`` or ``disabled``.
```

Comment:
```suggestion
with either of these 2 options set as ``false`` or ``disabled``.
```
What about KPR in partial mode?

Reply:
What feature does it need exactly? Can kube-proxy replacement be used in partial mode, or does it have to be strict mode?

Reply:
This feature is designed and tested under kube-proxy-replacement strict mode. I don't know if we can support its functionality when kube-proxy is partially present. @anfernee could you comment on this?

Reply:
`=partial` won't work with the egress gateway unless all KPR sub-features are enabled, so I think we should just recommend to set `=strict`.